sdcflows/workflows/topup.py

```diff
@@ -0,0 +1,359 @@
+
+    This workflow takes in a set of EPI files with opposite phase encoding
+    direction than the target file and calculates a fieldmap (off-resonance field) in Hz that can
+    be fed into the ``fmap`` workflows
+
+    This procedure works if there is only one ``_epi`` file is present
+    (as long as it has the opposite phase encoding direction to the target
+    file). The target file will be used to estimate the field distortion.
+    However, if there is another ``_epi`` file present with a matching
+    phase encoding direction to the target it will be used instead.
+
+    Currently, different phase encoding directions in the target file and the
+    ``_epi`` file(s) (for example, ``i`` and ``j``) is not supported.
+
+    The off-resonance field is estimated using FSL's ``Topup``. Topup also calculates undistorted
+    versions of the inputted EPI files which can be used as magnitute input in the ``fmap``
+    worklows.
+
+    Workflow Graph
+        .. workflow ::
+            :graph2use: orig
+            :simple_form: yes
+
+            from sdcflows.workflows.topup import init_topup_wf
+            wf = init_topup_wf()
+
+    Parameters
+    ----------
+    matched_pe : bool
+        Whether the input ``fmaps_epi`` will contain images with matched
+        PE blips or not. Please use :func:`sdcflows.workflows.topup.check_pes`
+        to determine whether they exist or not.
+    name : str
+        Name for this workflow
+    omp_nthreads : int
+        Parallelize internal tasks across the number of CPUs given by this option.
+
+    Inputs
+    ------
+    fmaps_epi : list of tuple(pathlike, str)
+        The list of EPI images that will be used in PE-Polar correction, and
+        their corresponding ``PhaseEncodingDirection`` metadata.
+        The workflow will use the ``epi_pe_dir`` input to separate out those
+        EPI acquisitions with opposed PE blips and those with matched PE blips
+        (the latter could be none, and ``in_reference`` would then be
+        used). The workflow raises a ``ValueError`` when no images with
+        opposed PE blips are found.
+    epi_pe_dir : str
+        The baseline PE direction.
+    epi_trt: float
+        Total readout time of the EPI (should be identical to TRT of fmaps_epi)
+    matched_pe_dir : str
+        Phase encoding direction of matching fmap EPI
+    opposed_pe_dir : str
+        Phase encoding direction of opposed fmap EPI
+    in_reference : pathlike
+        The baseline reference image (must correspond to ``epi_pe_dir``).
+
+    Outputs
+    -------
+    fieldmap : pathlike
+        Topup estimated fieldmap (Hz)
+    magnitude : pathlike
+        The ``fmaps_epi`` after unwarping
+    """
+
+    workflow = Workflow(name=name)
+    workflow.__desc__ = """\
+A B0-nonuniformity map (or *fieldmap*) was estimated based on two (or more)
+echo-planar imaging (EPI) references with opposing phase-encoding
+directions, with `Topup`).
```

Comment:
```suggestion
echo-planar imaging (EPI) references of opposed phase-encoding
directions with `topup` @topup (FSL {fsl_ver}).
```

We will need to add a topup entry to the bib file, referencing
> J.L.R. Andersson, S. Skare, J. Ashburner. How to correct susceptibility distortions in spin-echo echo-planar images: application to diffusion tensor imaging. NeuroImage, 20(2):870-888, 2003.

Reply:
FSL version gets returned in #106, but not sure where I find the bib-file?